ALPHA_BETA.md

```diff
@@ -0,0 +1,109 @@
+# rubot: alpha_beta
+
+The goal of the `Bot` is to take a decision tree and output
+the optimal decision.
+
+At each depth of this tree we either want to maximise or minimize
+the resulting fitness.
+
+In a competitive two player game, maximizing depths are the turns in which the bot
+is active, and during minimizing depths the opponent is able to make a decision.
+
+- `a n` := the bot is active, meaning we want to take the path which results in the highest fitness.
+    the node itself has a fitness of `n`.
+- `o n` := the opponent is active, meaning that we want to achieve the lowest possible fitness.
+
+Paths are represented as a list of indices.
+
+## Examples
+
+```
+a 0
+  - a 12 # child 0
+  - a 7  # child 1
+
+optimal action: [0], with fitness: 12
```

Comment:
I would include an example of how we calculate this along with some examples of changing something while leaving everything else in tact and observing the results. 

I.e. for this example fitness of the root node is irrelevant (if we change it to -1 or to 1e6) the result is still 12. And we find the maximum of the fitnesses of all children. In this case it's also irrelevant whether children are `a` or `o`, since fitness for terminal nodes are just their values or both `a` and `o`.

It's also interesting that you don't interchange `a` in `o` as usually done in game tree (since opponents usually alternate moves). But I think it's fine, there might be some games with weird move order and one can generalise to that.